kyaml/kio/byteio_reader.go

```diff
@@ -119,7 +122,14 @@ func (r *ByteReader) Read() ([]*yaml.RNode, error) {
 
 	index := 0
 	for i := range values {
-		decoder := yaml.NewDecoder(bytes.NewBufferString(values[i]))
+		value := values[i]
+		if strings.HasSuffix(r.Path, JSONSuffix) {
```

Comment:
We probably need to also update the package write to write json as well?

Reply:
Done.